### PR TITLE
fix: clear all console warnings

### PR DESCRIPTION
**Pull request by sh1luo**
Signed-off-by: sh1luo <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
Before:
image

After:
image

change `a` to `button` and add css style, referer [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md#case-i-understand-the-previous-cases-but-still-need-an-element-resembling-a-link-that-is-purely-clickable)

File: web/src/auth/DingTalkLoginButton.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/dingtalk.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/dingtalk.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/DingTalkLoginButton.js:19]: Set something (like "dingtalk") to alt.

File: web/src/auth/FacebookLoginButton.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/facebook.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/facebook.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/FacebookLoginButton.js:19]: Set something to alt.

File: web/src/auth/GitHubLoginButton.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/github.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/github.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/GitHubLoginButton.js:19]: Set something to alt.

File: web/src/auth/GiteeLoginButton.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/gitee.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/gitee.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/GiteeLoginButton.js:19]: Set something to alt.

File: web/src/auth/GoogleLoginButton.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/google.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/google.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/GoogleLoginButton.js:19]: Set something to alt.

File: web/src/auth/WechatLoginButton.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/wechat.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/wechat.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/WechatLoginButton.js:19]: Set something to alt.

File: web/src/auth/WeiboLoginButton.js [hunk 7]
```diff
@@ -16,7 +16,7 @@ import {createButton} from "react-social-login-buttons";
 import {StaticBaseUrl} from "../Setting";
 
 function Icon({ width = 24, height = 24, color }) {
-    return <img src={`${StaticBaseUrl}/buttons/weibo.svg`} />;
+    return <img src={`${StaticBaseUrl}/buttons/weibo.svg`} alt=""/>;
```
Comment (hsluoyz) [inline web/src/auth/WeiboLoginButton.js:19]: Set something to alt.

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Demo link?

**Comment by sh1luo** [comment]
> Demo link?

I think there is no need for a demo link here, just clone the pr , `npm run start`,

**Comment by hsluoyz** [comment]
@sh1luo plz provide one

**Comment by sh1luo** [comment]
@hsluoyz demo link: http://1.13.13.178:8000/

File: web/src/auth/LoginPage.js [hunk 8]
```diff
@@ -283,11 +283,11 @@ class LoginPage extends React.Component {
                 {i18next.t("login:Auto login")}
               </Checkbox>
             </Form.Item>
-            <a style={{float: "right"}} onClick={() => {
+            <button style={{float: "right"}} className="link-button" onClick={() => {
```
Comment (hsluoyz) [inline web/src/auth/LoginPage.js:286]: Your code doesn't have hyperlink highlight, see: https://door.casbin.com/login

image

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/auth/LoginPage.js (hunk 8 above)
Comment (sh1luo) [reply web/src/auth/LoginPage.js:286]: Already fixed

Comment (hsluoyz) [reply web/src/auth/LoginPage.js:286]: @sh1luo  Harding-coding the colors is not the correct way. Plz rollback these a tag changes.

**Pull request merged**